app/controllers/deceased_settlor/mld5/CountryOfResidenceController.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.deceased_settlor.mld5
+
+import config.annotations.DeceasedSettlor
+import controllers.actions._
+import controllers.actions.deceased_settlor.NameRequiredActionProvider
+import forms.CountryFormProvider
+import javax.inject.Inject
+import models.Mode
+import navigation.Navigator
+import pages.deceased_settlor.SettlorsNamePage
+import pages.deceased_settlor.mld5.CountryOfResidencePage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.RegistrationsRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import utils.countryOptions.CountryOptionsNonUK
+import views.html.deceased_settlor.mld5.CountryOfResidenceView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CountryOfResidenceController @Inject()(
+                                                   override val messagesApi: MessagesApi,
+                                                   registrationsRepository: RegistrationsRepository,
+                                                   @DeceasedSettlor navigator: Navigator,
+                                                   actions: Actions,
+                                                   requireName: NameRequiredActionProvider,
+                                                   formProvider: CountryFormProvider,
+                                                   val controllerComponents: MessagesControllerComponents,
+                                                   view: CountryOfResidenceView,
+                                                   val countryOptions: CountryOptionsNonUK
+                                 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  val form: Form[String] = formProvider.withPrefix("5mld.countryOfResidence")
+
+  def onPageLoad(mode: Mode, draftId: String): Action[AnyContent] = (actions.authWithData(draftId) andThen requireName(draftId)) {
+    implicit request =>
+
+      val name = request.userAnswers.get(SettlorsNamePage).get
```

Comment:
Having done requireName can you not retrieve the name from the request?

Reply:
Updated controllers